kinbot/uncertainty_analysis.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import division
+from __future__ import print_function
+import os
+import random
+import numpy as np
+
+"""
+Functions for the generation and postprocessing of uncertainty analysis
+"""
+
+
+class UQ:
+
+    def __init__(self, par):
+        self.par = par
+        self.wellUQ = par['well_uq']
+        self.barUQ = par['barrier_uq']
+        self.freqUQ = par['freq_uq']
+        self.imagfreqUQ = par['imagfreq_uq']
+        self.hirUQ = par['hir_uq']
+        self.relaxPowerUQ = par['energy_relaxation_power_uq']
+        self.relaxFactorUQ = par['energy_relaxation_factor_uq']
+        self.eWellUQ = par['epsilon_well_uq']
+        self.sWellUQ = par['sigma_well_uq']
+        self.uq_iter = 0
+
+    def calc_rotor_factor(self, propertyType, species, uq_iter, runUQ, name):
+        string = []
+        for i, char in enumerate(name):
+            string.append(char)
+        string.reverse()
+        stop_parsing = 0
+        name_array = []
+        for char in string:
+            if stop_parsing == 0:
+                if char != " ":
+                    name_array.append(char)
+                elif char == " ":
+                    stop_parsing = 1
+            else:
+                pass
+        name_array.reverse()
+        name_string = ''.join(name_array)
+
+        if uq_iter == 0:
+            factor = 1
+            normfactor = 0
+
+            return factor, normfactor
+        else:
+            factor = np.exp(random.uniform(np.log(1./self.hirUQ), np.log(self.hirUQ)))
+            normfactor = np.log(factor) / np.log(self.hirUQ)
+            if normfactor < -1 or normfactor > 1:
+                e = "ERROR"
+            else:
+                e = ''
+
+        return factor, normfactor
+
+    def calc_factor(self, propertyType, species, uq_iter, runUQ):
```

Comment:
the `runUQ` parameter is meant to establish if the UQ values should be set yet. For instance, in a PES run, we need to wait to calculate the values until we get past all of the KinBot steps and piece together all of the mess files from each kinbot run.

In the case of a KinBot run, the mess files can be generated when the KinBot run is done because we don't need to account for piecing a MESS file together from multiple kinbot runs at once. 

Basically if `runUQ=0` then place holders for energies are set during kinbot runs, and are filled in later once all the mess files (with place holders) have been created.

If `runUQ=1` then the mess files are created wth the UQ values input into them immediately without any place holders.